src/client/src/Stream.c

```diff
@@ -812,10 +816,11 @@ STATUS putFrame(PKinesisVideoStream pKinesisVideoStream, PFrame pFrame)
     }
 
     // Check if the frames should be skipped and whether the skip flag needs to be reset
+    // If current frame is EOFR, We will package EOFR frame and we do not need to reset skip flag
     if (pKinesisVideoStream->skipNonKeyFrames) {
         if (CHECK_FRAME_FLAG_KEY_FRAME(pFrame->flags)) {
             pKinesisVideoStream->skipNonKeyFrames = FALSE;
-        } else {
+        } else if(!CHECK_FRAME_FLAG_END_OF_FRAGMENT(pFrame->flags)) {
```

Comment:
If you set the skipNonKeyFrames after you process the eofr you won't need this if clause which is harder to understand.

Reply:
It's not the same thing right, it's possible pKinesisVideoStream->skipNonKeyFrames was already TRUE without me setting it here, in that case the else would not do the same thing.  

I think this is now easier to understand because I'm being explicit about skipping only if its not an EOFR (and also not a key frame).

Reply:
Are you saying that we're relying on `pKinesisVideoStream->skipNonKeyFrames` being FALSE?
At stream creation the value is TRUE.  As soon as the first key-frame is encountered it becomes FALSE.  The only way it goes back to TRUE is if we have a bad fragment or during the executeStoppedStreamState if there's a timeout and no data available. 

Even if we could rely on it (which I'm not sure we can) I still think it's better to be clear about setting it in the beginning and being clear about skipping non key frames that come after an EOFR.